### PR TITLE
LPD-18486 Manage addresses on account contacts

### PR DESCRIPTION
**Pull request by dsitu**
https://liferay.atlassian.net/browse/LPD-18486

### CODE REVIEW

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
@dsitu wrong commit names, it's LPD-18486 not LPD-18468 please rename. Thanks

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	@Override
+	protected Address construct(ActionRequest actionRequest) throws Exception {
+		String street1 = ParamUtil.getString(actionRequest, "addressStreet1");
+		String street2 = ParamUtil.getString(actionRequest, "addressStreet2");
+		String street3 = ParamUtil.getString(actionRequest, "addressStreet3");
+		String city = ParamUtil.getString(actionRequest, "addressCity");
+		String zip = ParamUtil.getString(actionRequest, "addressZip");
+		long countryId = ParamUtil.getLong(actionRequest, "addressCountryId");
```
Comment (luca-pellizzon) [inline modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java:41]: either you sort them or you name them as the params

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			ParamUtil.getLong(actionRequest, "addressRegionId"));
+		address.setCity(city);
+		address.setMailing(
+			ParamUtil.getBoolean(actionRequest, "addressMailing"));
+		address.setPrimary(
+			ParamUtil.getBoolean(actionRequest, "addressPrimary"));
+		address.setStreet1(street1);
+		address.setStreet2(street2);
+		address.setStreet3(street3);
+		address.setZip(zip);
```
Comment (luca-pellizzon) [inline modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java:67]: sort

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java (hunk 1 above)
Comment (smottal) [reply modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java:41]: I vote for the same name as the params

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java (hunk 2 above)
Comment (smottal) [reply modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/manager/AddressContactInfoManager.java:67]: is this the same order as in the service.xml? If yes I think it's fine

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/account_contact/edit_address.jsp [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+if (editContactInformationDisplayContext.getPrimaryKey() > 0) {
+	address = AddressServiceUtil.getAddress(editContactInformationDisplayContext.getPrimaryKey());
+
+	countryId = address.getCountryId();
+	regionId = address.getRegionId();
+}
+
+PortalUtil.addPortletBreadcrumbEntry(request, LanguageUtil.get(request, "addresses"), editContactInformationDisplayContext.getRedirect());
+
```
Comment (smottal) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/account_contact/edit_address.jsp:28]: Based on this https://github.com/liferay/liferay-portal/commit/2445f2b8ba8b3eb037885e4d36cef3417fc1dc8d this line should be removed

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/account_contact/edit_address.jsp [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	}
+
+	if (addressCountry) {
+		addressCountry.addEventListener('change', handleSelectChange);
+
+		<c:if test="<%= countryId > 0 %>">
+			checkCountry(<%= countryId %>);
+		</c:if>
+	}
+</aui:script>
```
Comment (smottal) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/account_contact/edit_address.jsp:208]: Can we put this in a separated js file? I think we are removing the scripts inside jsp

File: modules/test/playwright/tests/account-admin-web/accounts.spec.ts [hunk 5]
```diff
@@ -63,6 +63,38 @@ test('LPD-18485 Update account contact information fields', async ({
...
+		await (await accountsPage.accountsTableRowLink(account.name)).click();
+		await editAccountPage.contactLink.click();
+		await accountContactAddressPage.addAddressesButton.click();
+		await editAccountContactAddressPage.updateAddress('address1', 'city');
+
+		await expect(
+			page.getByText('Success:Your request completed successfully.')
+		).toBeVisible();
+
+		await expect(page.getByText('address1city,')).toBeVisible();
```
Comment (smottal) [inline modules/test/playwright/tests/account-admin-web/accounts.spec.ts:91]: can you create a dynamic locator (so a function that accept the name) inside the page? And is it the `,` needed?

File: modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v2_10_4/AccountListTypeUpgradeProcess.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @author Danny Situ
+ */
+public class AccountListTypeUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		runSQL(
+			StringBundler.concat(
+				"delete from ListType where type_ ='",
```
Comment (smottal) [inline modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v2_10_4/AccountListTypeUpgradeProcess.java:21]: please add a space between `=` and `'`

File: modules/apps/account/account-service/bnd.bnd
Comment (smottal) [inline modules/apps/account/account-service/bnd.bnd:1]: Can you move this file on the next commit? It's also related to baseline

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by smottal** [comment]
Danny please have also a look to the comments of BChan on the previous PR if it's something that we re-do in this one. Thx

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Review by smottal — APPROVED** [review]

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Review by luca-pellizzon — APPROVED** [review]

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**